permissions_validators/src/private_blockchain/mod.rs

```diff
@@ -38,3 +38,421 @@ impl<W: WorldTrait> IsGrantAllowed<W> for ProhibitGrant {
         Err("Granting at runtime is prohibited.".to_owned())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #![allow(clippy::restriction)]
+
+    use std::str::FromStr as _;
+
+    use super::*;
+
+    type TestEnv = (
```

Comment:
Doesn't matter here, but this should be a `struct`. You can bind directly to names using Rust's pattern matching syntax.